reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md

```diff
@@ -175,19 +175,23 @@ function Get-FunctionPosition {
   process {
     try {
       $filesToProcess = if ($_ -is [System.IO.FileSystemInfo]) {
+        Write-Verbose "From pipeline"
         $_
       } else {
-        $filesToProcess = Get-Item -Path $Path
+        Write-Verbose "From parameter, $Path"
+        Get-Item -Path $Path
```

Comment:
Why did you remove the assignment here? If this code path is used then the collection in the foreach loop is empty.

Reply:
Because the **if** starts with `$filesToProcess = if (...` therefor inside the **else** script block you cannot assign value to the same variable.
I tested it and didn't work.

Reply:
DOH, I see it now. Thanks.